pkg/utils/db.go

```diff
@@ -524,3 +524,86 @@ func AddGSetWithPurged(ctx context.Context, gset gtid.Set, conn *sql.Conn) (gtid
 	_ = ret.Set(newGset)
 	return ret, nil
 }
+
+// AdjustSQLModeCompatible adjust downstream sql mode to compatible
+func AdjustSQLModeCompatible(sqlModes string) (string, error) {
+	needDisable := []string{
+		"NO_ZERO_IN_DATE",
+		"NO_ZERO_DATE",
+		"ERROR_FOR_DIVISION_BY_ZERO",
+		"NO_AUTO_CREATE_USER",
+		"STRICT_TRANS_TABLES",
+		"STRICT_ALL_TABLES",
+	}
+	needEnable := []string{
+		"IGNORE_SPACE",
+		"NO_AUTO_VALUE_ON_ZERO",
+		"ALLOW_INVALID_DATES",
+	}
+	disable := strings.Join(needDisable, ",")
+	enable := strings.Join(needEnable, ",")
+
+	mode, err := tmysql.GetSQLMode(sqlModes)
+	if err != nil {
+		return "", err
+	}
+	disableMode, err2 := tmysql.GetSQLMode(disable)
+	if err2 != nil {
+		return "", err2
+	}
+	enableMode, err3 := tmysql.GetSQLMode(enable)
+	if err3 != nil {
+		return "", err3
+	}
+	mode = (mode &^ disableMode) | enableMode
```

Comment:
could you please explain the result after this bit manipulation?

Reply:
Here, disableMode contains all sqlMode bits that need to be deleted, and enableMode contains all sqlMode bits that need to be added. Then use the upstream's mode NAND disableMode, and then OR with enableMode to get the result we want. For example
```txt
 (0110 &^ 0011) | 1000 
= 0100 | 1000
= 1100
```

Reply:
maybe add a comment to code